src/Symfony/Component/HttpKernel/DataCollector/RequestDataCollector.php

```diff
@@ -170,27 +170,27 @@ public function getRequestQuery()
 
     public function getRequestHeaders()
     {
-        return new HeaderBag($this->data['request_headers']);
+        return new ParameterBag($this->data['request_headers']);
```

Comment:
I was more thinking about references in core templates ;) but i guess were fine.